QueryBuilder.Test/Where.UnitTests.cs

```diff
@@ -153,6 +154,56 @@ public void CanApplyNotInFilters()
             Assert.AreEqual($"SELECT bldng FROM DIGITALTWINS bldng JOIN itfunc RELATED bldng.hasITSiteFunction rel WHERE IS_OF_MODEL(bldng, '{Building.ModelId.UpdateVersion(1)}') AND IS_OF_MODEL(itfunc, '{ITSiteFunction.ModelId.UpdateVersion(1)}') AND rel.maxPriority NIN ['name1','name2'] AND bldng.$dtId = 'ID'", query.BuildAdtQuery());
         }
 
+        [TestMethod]
+        public void CanApplyWhereIsOfModelFilter()
+        {
+            var query = QueryBuilder
+                    .From<Space>()
+                    .Or(query => query
+                        .WhereIsOfModel<Space, Building>()
+                        .WhereIsOfModel<Space, Floor>());
+
+            Assert.AreEqual($"SELECT space FROM DIGITALTWINS space WHERE IS_OF_MODEL(space, '{Space.ModelId.UpdateVersion(1)}') AND (IS_OF_MODEL(space, '{Building.ModelId.UpdateVersion(1)}') OR IS_OF_MODEL(space, '{Floor.ModelId.UpdateVersion(1)}'))", query.BuildAdtQuery());
```

Comment:
`IS_OF_MODEL(space, '{Space.ModelId.UpdateVersion(1)}')` should not be added here. This reduncancy issue is different from the duplication issue of users invoking IsOfModel with the same params twice.
A user adding a type filer on the root twin is basically trying to replace the default filter - same for the join type filters

Reply:
As per our discussion, let's not handle this case. Handling this properly would mean interfering with the semantics of the query and only remove the auto-added filter when it is implied by the user-added ones. 
This issue is only of concern if the users add a model filter while querying FROM a specific model, where the standard is that users should only add model filters if they're selecting FROM `BasicDigitalTwins`. 

So, @Isai-Cicourel, please add explicit recommendation in the documentation to only add a model filter when there is no model inferred from the FROM clause, i.e. use `From<BasicDigitalTwin>` if they plan on adding IsOfModel filter.